.github/container/test-maxtext.sh

```diff
@@ -139,26 +158,39 @@ done
 
 # # Set derived variables
 
+# if not the gpt3-52k, we can use any any attention type such as cudnn_flash_te or dot_product
+if [ $MODEL != "gpt3-52k" ]; then # gpt3-52k only works with dot_product
+    ADDITIONAL_ARGS+=" attention=${ATTN_TYPE}"
+fi
+
+# for fp8 runs
+if [ $DTYPE == "fp8" ]; then # gpt3-52k only works with dot_product
+    ADDITIONAL_ARGS+=" quantization=$DTYPE"
+fi
+
 GPUS_PER_NODE=$(nvidia-smi -L | grep -c '^GPU')
 NGPUS=$((GPUS_PER_NODE * NODES))
 
 # Heuristic to figure out ici and dcn of DP
-# We only use DP across different nodes
-if [ $NGPUS -gt 8 ]; then
-    dcn_DP=$((NGPUS / 8))
-    ici_DP=$((DP / dcn_DP))
+# TP is always ici; after TP it will be FSDP and
+# from TP and FSDP, we can find out ici and dcn DP
+# in other words, DP dim across ici and dcn axis will always be inferred
+ici_TP=${TP}
+ici_DP=1
+dcn_FSDP=1
+if [ $((FSDP*TP)) -gt 8 ]; then
```

Comment:
Maybe add a (c) note that the script expect 8 GPUs per node.
If needed, later someone can lift.

Reply:
good point, I can actually use the variable `GPUS_PER_NODE=$(nvidia-smi -L | grep -c '^GPU')` then it should be alright